packages/fast-components-react-base/src/listbox/listbox.tsx

```diff
@@ -0,0 +1,314 @@
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { ListboxClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    ListboxHandledProps,
+    ListboxProps,
+    ListboxUnhandledProps,
+} from "./listbox.props";
+import * as React from "react";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import { get, inRange, invert } from "lodash-es";
+import { canUseDOM } from "exenv-es6";
+
+export interface ListboxState {
+    /**
+     * The index of the focusable child
+     */
+    focusIndex: number;
+}
+
+export interface TypeAheadDataItem {
+    index: number;
+    compareString: string;
+}
+
+class Listbox extends Foundation<
+    ListboxHandledProps,
+    ListboxUnhandledProps,
+    ListboxState
+> {
+    public static displayName: string = "Listbox";
+
+    protected handledProps: HandledProps<ListboxHandledProps> = {
+        children: void 0,
+        managedClasses: void 0,
+        typeAheadPropName: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    private typeAheadString: string = "";
+    private typeAheadTimer: any;
+    private typeAheadData: TypeAheadDataItem[] = [];
+
+    constructor(props: ListboxProps) {
+        super(props);
+
+        this.state = {
+            focusIndex: -1,
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                ref={this.rootElement}
+                role="listbox"
+                className={this.generateClassNames()}
+                onKeyDown={this.handleMenuKeyDown}
+            >
+                {this.renderChildren()}
+            </div>
+        );
+    }
+
+    public componentDidMount(): void {
+        const children: Element[] = this.domChildren();
+        const focusIndex: number = children.findIndex(this.isFocusableElement);
+
+        if (focusIndex !== -1) {
+            this.setState({
+                focusIndex,
+            });
+        }
+    }
+
+    public componentWillUnmount(): void {
+        clearTimeout(this.typeAheadTimer);
+    }
+
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        return super.generateClassNames(get(this.props.managedClasses, "listbox"));
+    }
+
+    /**
+     * Render all child elements
+     */
+    private renderChildren(): React.ReactChild[] {
+        this.typeAheadData = [];
+        return React.Children.map(this.props.children, this.renderChild);
+    }
+
+    /**
+     * Render a single child
+     */
+    private renderChild = (
+        child: React.ReactElement<any>,
```

Comment:
I wonder if this is better typed as `unknown` instead of `any`

Reply:
I get errors when make that change.